projects/core/directives/dropdown/dropdown-hover.directive.ts

```diff
@@ -30,7 +44,28 @@ export class TuiDropdownHover extends TuiDriver {
     private readonly options = inject(TUI_DROPDOWN_HOVER_OPTIONS);
     private readonly activeZone = inject(TuiActiveZone);
     private readonly open = inject(TuiDropdownOpen, {optional: true});
+    /**
+     * Dropdown can be removed not only via click/touch –
+     * swipe on mobile devices removes dropdown sheet without triggering new mouseover / mouseout events.
+     */
+    private readonly dropdownExternalRemoval$ = toObservable(
+        inject(TuiDropdownDirective).ref,
+    ).pipe(
+        filter((x) => !x && this.hovered),
+        share(),
+    );
+
     private readonly stream$ = merge(
+        this.dropdownExternalRemoval$.pipe(map(() => null)),
+        this.dropdownExternalRemoval$.pipe(
+            switchMap(() =>
+                tuiTypedFromEvent(this.doc, 'pointerdown').pipe(
+                    map(tuiGetActualTarget),
+                    delay(this.hideDelay),
+                    takeUntil(fromEvent(this.doc, 'mouseover')),
+                ),
+            ),
+        ),
```

Comment:
Guys, should I replace it by the following code ? 🤔 

```suggestion
    private readonly stream$ = merge(
        this.dropdownExternalRemoval$.pipe(
            switchMap(() =>
                tuiTypedFromEvent(this.doc, 'pointerdown').pipe(
                    map(tuiGetActualTarget),
                    delay(this.hideDelay),
                    takeUntil(fromEvent(this.doc, 'mouseover')),
                ),
            ),
            startWith(null),
        ),
```

Reply:
\+ remove `share()`